pkg/collector/listeners/docker.go

```diff
@@ -37,6 +37,7 @@ const (
 // match templates against.
 type DockerListener struct {
 	Client     *client.Client
+	docker     *docker.DockerUtil
```

Comment:
it's the same name has the package, it might be confusing. Maybe we should rename it `dockerUtil`.

Also should we store the DockerUtil object ? I thought we should always call `GetDockerUtil` in case we were disconnected or something like that ? `DockerService` seems to to that.

Reply:
`docker` is not ideal, but `dockerutil` seemed too long and `du` not expressive enough. But I'll go with `dockerUtil` indeed.

The `Retrier` is only handling errors on init (in case the service starts after the agent), not afterwards (once status == OK it does nothing at all). We can safely require an agent restart after  a config change in docker/kubelet. Furthermore, the next PR will be refactoring the event streaming logic into `DockerUtil` itself, so `DockerListener` will keep two open channels with the utils, might as well keep the object itself.